python/cudf/cudf/core/buffer/exposure_tracked_buffer.py

```diff
@@ -139,103 +41,50 @@ def mark_exposed(self) -> None:
         self._exposed = True
 
     @classmethod
-    def _from_device_memory(cls, data: Any, *, exposed: bool = False) -> Self:
-        """Create an exposure tracked buffer from device memory.
-
-        No data is being copied.
-
-        Parameters
-        ----------
-        data : device-buffer-like
-            An object implementing the CUDA Array Interface.
-        exposed : bool, optional
-            Mark the buffer as permanently exposed.
-
-        Returns
-        -------
-        ExposureTrackedBuffer
-            Buffer representing the same device memory as `data`
-        """
-        ret = super()._from_device_memory(data)
+    def _from_device_memory(cls, data: Any, exposed: bool) -> Self:
+        ret = super()._from_device_memory(data, exposed=exposed)
         ret._exposed = exposed
         ret._slices = weakref.WeakSet()
         return ret
 
-    def _getitem(self, offset: int, size: int) -> BufferSlice:
-        return BufferSlice(base=self, offset=offset, size=size)
-
     @property
     def __cuda_array_interface__(self) -> Mapping:
         self.mark_exposed()
```

Comment:
If a `BufferOwner` offered `exposed` and `mark_exposed` (that were respectively `False` and a no-op) then there wouldn't be the need for a separate implementation of `__cuda_array_interface__` here, and I think the only "special" code would have to be `_from_device_memory` and the implemenation of `mark_exposed`.

Like this?

```python
class Owner:

    mem: DeviceBuffer
    exposed: bool
    _slices: weakref.WeakSet

    def mark_exposed(self):
        pass

    @classmethod
    def _from_device_memory(cls, mem, exposed):
        self = cls.__new__(cls)
        self.mem = mem
        self.exposed = False
        return self

        
    @property
    def __cuda_array_interface__(self):
        self.mark_exposed()
        return ...


class ExposedTracker(Owner):
    def mark_exposed(self):
        self.exposed = True

    @classmethod
    def _from_device_memory(cls, mem, exposed):
        self = super()._from_device_memory(mem, exposed)
        self.exposed = exposed
        self._slices = weakref.WeakSet()
        return self
```

Reply:
good idea, done

Reply:
I'm somewhat second-guessing the benefit of even separating exposure tracking from the parent BufferOwner class. If the parent needs to define the exposed property already for Liskov substitutability, then perhaps the two classes should just be merged (I made the same point above but reiterating in this thread). What if we did that, and it was only the usage of non-owning ExposureTrackedBuffer vs the Buffer that differentiated COW mode from non-COW mode?

Reply:
`ExposureTrackedBufferOwner` has been merged with `BufferOwner`